demosplan/DemosPlanCoreBundle/Repository/ProcedureRepository.php

```diff
@@ -1172,34 +1180,48 @@ public function getProcedures(array $procedureIds): array
     }
 
     /**
-     * @param bool $internal
-     * @param bool $asArray
+     * @return array<int, Procedure>
      *
-     * @return Procedure[]|array[]
+     * @throws PaginationException
+     * @throws PathException
+     * @throws SortException
+     */
+    public function getUndeletedProcedures(): array
+    {
+        $query = $this->createFluentQuery();
+        $query->getConditionDefinition()
+            ->propertyHasValue(false, ['deleted'])
+            ->propertyHasValue(false, ['master'])
+            ->propertyHasValue(false, ['masterTemplate']);
+
+        return $query->getEntities();
+    }
+
+    /**
+     * @return array<int, Procedure>
      *
      * @throws Exception
      */
-    public function getProceduresWithEndedParticipation(array $phaseKeys, $internal = true, $asArray = false): array
+    public function getProceduresWithEndedParticipation(array $phaseKeys, bool $internal = true): array
     {
         try {
             $currentDate = new DateTime();
-            $endDate = $internal ? 'procedure.endDate' : 'procedure.publicParticipationEndDate';
-            $phase = $internal ? 'procedure.phase' : 'procedure.publicParticipationPhase';
+            $procedures = $this->getUndeletedProcedures();
```

Comment:
This will most probably lead to a performance hit but as this is only used in maintenance task this may be feasible

Reply:
Yes I thought about it and decided to pay the price of the performance to improve readability and be able to handle objects. Out of this: I assume there will be only a few procedures which needs to be handled/filtered, therefore the impact on the performance isn't even measurable.